src/common/utils/router-context/index.ts

```diff
@@ -0,0 +1,43 @@
+import { qs } from 'url-parse';
+
+import { DEFAULT_MATCH, DEFAULT_ROUTE } from '../../constants';
+import {
+  Location,
+  MatchParams,
+  Query,
+  Route,
+  Routes,
+  RouterContext,
+} from '../../types';
+import matchRoute from '../match-route';
+import generatePath from '../generate-path';
+
+export const createRouterContext = (
+  route: Route,
+  params: MatchParams = {},
+  query: Query = {}
+): RouterContext => {
+  const pathname = generatePath(route.path, params);
+  const matchedRoute = matchRoute([route], pathname, query);
+
+  return {
+    route,
+    match: matchedRoute ? matchedRoute.match : DEFAULT_MATCH,
+    query,
+  };
+};
+
+export const findRouterContext = (
```

Comment:
This is `getRouteContext` renamed

Reply:
y tho?

Reply:
Because it was wrong: we have both `RouterContext` and `RouteContext` as types. `RouterContext` now is just `route`, `match` and `query`, while `RouteContext` has also `location` and `action`. But `getRouteContext` never returned a valid `RouteContext` (as was missing `action`). It was returning an old `RouterContext` (with `location`), so it should have been named `getRouterContext` to be correct. So given the name should be fixed anyway, I'm making it more explicit given it accepts an array of routes as input.